### ISSUE TITLE
error: identifier "__ldg" is undefined on pre-release branch.

### ISSUE THREAD

**Issue by neoblizz** [gunrock/gunrock#192]
**deviceQuery:**
```
Device 0: "GeForce GTX 970"
  CUDA Driver Version / Runtime Version          7.5 / 7.5
  CUDA Capability Major/Minor version number:    5.2
  Total amount of global memory:                 4096 MBytes (4294770688 bytes)
  (13) Multiprocessors, (128) CUDA Cores/MP:     1664 CUDA Cores
  GPU Max Clock rate:                            1178 MHz (1.18 GHz)
  Memory Clock rate:                             3505 Mhz
  Memory Bus Width:                              256-bit
  L2 Cache Size:                                 1835008 bytes
  Maximum Texture Dimension Size (x,y,z)         1D=(65536), 2D=(65536, 65536), 3D=(4096, 4096, 4096)
  Maximum Layered 1D Texture Size, (num) layers  1D=(16384), 2048 layers
  Maximum Layered 2D Texture Size, (num) layers  2D=(16384, 16384), 2048 layers
  Total amount of constant memory:               65536 bytes
  Total amount of shared memory per block:       49152 bytes
  Total number of registers available per block: 65536
  Warp size:                                     32
  Maximum number of threads per multiprocessor:  2048
  Maximum number of threads per block:           1024
  Max dimension size of a thread block (x,y,z): (1024, 1024, 64)
  Max dimension size of a grid size    (x,y,z): (2147483647, 65535, 65535)
  Maximum memory pitch:                          2147483647 bytes
  Texture alignment:                             512 bytes
  Concurrent copy and kernel execution:          Yes with 2 copy engine(s)
  Run time limit on kernels:                     No
  Integrated GPU sharing Host Memory:            No
  Support host page-locked memory mapping:       Yes
  Alignment requirement for Surfaces:            Yes
  Device has ECC support:                        Disabled
  Device supports Unified Addressing (UVA):      Yes
  Device PCI Domain ID / Bus ID / location ID:   0 / 2 / 0
  Compute Mode:
     < Default (multiple host threads can use ::cudaSetDevice() with device simultaneously) >
```

**cmake**
```
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found CUDA: /usr/local/cuda-7.5 (found suitable version "7.5", minimum required is "7.5")
-- Boost version: 1.58.0
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   timer
--   chrono
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp
-- Found OpenMP
-- Found Metis
-- Configuring done
-- Generating done
-- Build files have been written to: /home/muhammad/gunrock/build
```
[make -j8.log](https://github.com/gunrock/gunrock/files/485462/make.out.txt)

**Comment by neoblizz** [comment]
`dev_debug` and `dev` branches build and run file on my machine, meeting all the requirements. And I forgot to mention, I am using compute requirements 3.0 and 3.5 (6 is turned off).

**Comment by yzhwang** [comment]
Hi @neoblizz according to this answer:
http://stackoverflow.com/questions/24069524/error-using-ldg-in-cuda-kernel-at-compile-time
If you use both compute capability 30 and 35 for compiling, this won't work. Also I think in dev_debug (and hence also in pre-release) we are using a wrapper to handle lower compute capability:
https://github.com/gunrock/gunrock/blob/dev_debug/gunrock/util/device_intrinsics.cuh#L85
So maybe change that __ldg to _ldg and see if that fixes the bug?

**Comment by jowens** [comment]
I haven't built this on OS X for a while, but this error is also now in OS X in the prerelease branch.

**Comment by jowens** [comment]
I confirm @yzhwang's checkin in https://github.com/gunrock/gunrock/commit/c3533093e84fd4df2934f9f10508c6ace76ef43a fixes the problem for OS X. Thanks, @yzhwang! You the man!

**Comment by neoblizz** [comment]
Fixed on Ubuntu 15.04, GTX 970 as well, thanks @yzhwang!
